app/controllers/idv/verify_info_controller.rb

```diff
@@ -0,0 +1,84 @@
+module Idv
+  class VerifyInfoController < ApplicationController
+    before_action :render_404_if_verify_info_controller_disabled
+    before_action :confirm_two_factor_authenticated
+    before_action :confirm_ssn_step_complete
+
+    def analytics_visited_event
+      :idv_doc_auth_verify_visited
+    end
+
+    def show
+      increment_step_counts
+      analytics.public_send(
+        analytics_visited_event, **analytics_arguments
+      )
+    end
+
+    private
+
+    def render_404_if_verify_info_controller_disabled
+      render_not_found unless IdentityConfig.store.doc_auth_verify_info_controller_enabled
+    end
+
+    def analytics_arguments
+      {
+        flow_path: flow_path,
+        step: 'verify',
+        step_count: current_flow_step_counts['verify'],
+        analytics_id: 'Doc Auth',
+        irs_reproofing: irs_reproofing,
+      }.merge(**extra_analytics_properties)
+    end
+
+    # copied from doc_auth_controller
+    def flow_session
+      user_session['idv/doc_auth']
+    end
+
+    def flow_path
+      flow_session[:flow_path]
+    end
+
+    def irs_reproofing
+      effective_user&.decorate&.reproof_for_irs?(
+        service_provider: current_sp,
+      ).present?
+    end
+
+    # Copied from capture_doc_flow.rb
+    # and from doc_auth_flow.rb
+    def extra_analytics_properties
+      capture_session_uuid = flow_session[:document_capture_session_uuid]
+      if capture_session_uuid
+        {
+          acuant_sdk_upgrade_ab_test_bucket:
+          AbTests::ACUANT_SDK.bucket(capture_session_uuid),
+        }
+      else
+        {}
+      end
+    end
+
+    # copied from verify_step
+    def pii
+      @pii = flow_session[:pii_from_doc]
+    end
+
+    # copied from address_controller
+    def confirm_ssn_step_complete
+      return if pii.present?
```

Comment:
I don't know if the `pii` method here exactly tells you the SSN has been entered. We may want to explicitly check that an SSN is included.

The `pii_from_doc` part of the flow session is, as the name suggests, PII read from the document (i.e. drivers license). The SSN gets _added_ to that when the user submits it. So the presence of PII from the document does not necessarily mean the SSN step is complete.

Reply:
good catch, updated in https://github.com/18F/identity-idp/pull/7558/commits/9a2f6b91f54f9c12386e81fbbc57ec941547620f